browser/src/core/Socket.js

```diff
@@ -1714,7 +1714,9 @@ app.definitions.Socket = L.Class.extend({
 		result.finish = function () {
 			if (this.active) {
 				var now = performance.now();
-				that.sendMessage('TRACEEVENT name=' + name + ' ph=X ts=' + Math.round(now * 1000) + ' dur=' + Math.round((now - this.begin) * 1000)
+				that.sendMessage('TRACEEVENT name=' + name +
+						 ' ph=X ts=' + Math.round(this.begin * 1000) +
+						 ' dur=' + Math.round((now - this.begin) * 1000)
```

Comment:
Good catch.

Reply:
Re: duplicating it - fair cop we could re-factor that but I want to add the timeout early to get in as fast as possible after we return from idle - but track the start time from the end of the method - could be done a nicer way I guess by updating 'begin' or something in an un-finished object - feel free to clean it up.
Thanks !